### PR TITLE
Submission viewer

### PR DESCRIPTION
**Pull request by BIG-COW**
- Replace pages aggregation_task and student_task to only one page.
- Allow to query submissions based on several criteria.
- Some changes in tags behavior.

### CODE REVIEW

**Comment by anthonygego** [comment]
Again, please tell us when you want this to be reviewed, as you're still pushing commits to the branch.

**Comment by BIG-COW** [comment]
Yes, you can review this. I should not commit anymore.

File: inginious/frontend/courses.py [hunk 1]
```diff
@@ -215,12 +216,35 @@ def get_all_tags_names_as_list(self, admin=False, language="en"):
...
+        
+            { 'org_tag_1': ['task_id', 'task_id', ...], 
+              'org_tag_2' : ['task_id', 'task_id', ...], 
+              ... }
+         """
+        if self._organisational_tags_to_task != {}:
+            return self._organisational_tags_to_task
+            
+        for org_tag in self.get_all_tags()[2]:
+            tag = org_tag.get_name()
```
Comment (anthonygego) [inline inginious/frontend/courses.py]: Is this gettext friendly ?

File: inginious/frontend/pages/course_admin/statistics.py [hunk 2]
```diff
@@ -14,6 +14,56 @@ def compute_statistics(task, data):
...
     )
      """
+    
+    super_dict = {}
+    for submission in data:
+        task = tasks[submission["taskid"]]
+        username = "".join(submission["username"])
+        tags_of_task = task.get_tags()[0] + task.get_tags()[1]
+        for tag in tags_of_task:
+            if tag not in super_dict:
```
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/statistics.py]: I think that the ``setdefault`` method can be useful here to make some clean up.

File: inginious/frontend/pages/course_admin/statistics.py [hunk 3]
```diff
@@ -14,6 +14,56 @@ def compute_statistics(task, data):
...
+                super_dict[tag] = {}
+            if username not in super_dict[tag]:
+                super_dict[tag][username] = {}
+            if submission["taskid"] not in super_dict[tag][username]:
+                super_dict[tag][username][submission["taskid"]] = [0,0,0,0]
+            super_dict[tag][username][submission["taskid"]][0] += 1
+            if "tests" in submission and tag.get_id() in submission["tests"] and submission["tests"][tag.get_id()] == True:
+                super_dict[tag][username][submission["taskid"]][1] += 1
+
+            if (submission["best"] == True):
```
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/statistics.py]: Parentheses not required. Is ``== True`` required ?

File: inginious/frontend/pages/course_admin/statistics.py [hunk 4]
```diff
@@ -14,6 +14,56 @@ def compute_statistics(task, data):
...
+                super_dict[tag][username][submission["taskid"]][2] += 1
+                if "tests" in submission and tag.get_id() in submission["tests"] and submission["tests"][tag.get_id()] == True:
+                    super_dict[tag][username][submission["taskid"]][3] += 1
+
+    output = []
+    for tag in super_dict:
+
+        #No ponderation
+        if ponderation == False:
```
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/statistics.py]: Again, is ``== False`` required ?

File: inginious/frontend/pages/course_admin/submission_viewer.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        return self.page(self.init(courseid))
+
+    def submission_url_generator(self, submissionid):
+        """ Generates a submission url """
+        return "?submission=" + submissionid
+
+    def page(self, course):
+        """ Get all data and display the page """
+        
+        input = self.get_input()
```
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/submission_viewer.py]: ``input`` is the built-in user input method in Python, the variable should be renamed.

File: inginious/frontend/pages/course_admin/submission_viewer.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+        data, classroom = self.get_submissions(course, input) #ONLY classrooms user wants to query
+        if len(data) == 0 and not self.show_collapse(input):
+            self._msg.append(_("No submissions found"))
+            
+        if "json" in input:
+            for submission in data:
+                submission = self.submission_manager.get_input_from_submission(submission)
+                #if need also feedback, uncomment this line
+                #submission = self.submission_manager.get_feedback_from_submission(submission,show_everything=True,translation=self.app._translations.get(self.user_manager.session_language(), gettext.NullTranslations()))
```
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/submission_viewer.py]: A choice has to be done.

File: inginious/frontend/pages/course_admin/submission_viewer.py [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+        users = self.get_users(course) # All users of the course
+        tasks = course.get_tasks();  # All tasks of the course
+
+        statistics = None
+        if input.stat != "no_stat":
+            old_time = time.time()
+            statistics = compute_statistics(tasks, data, True if "with_pond_stat" == input.stat else False)
+            total_time = time.time() - old_time
+            if total_time > self._time_alert_computation_stats:
+                self._msg.append(_("Statistics take {0} seconds to compute due to the higher number of submissions. Try adding more filter options to reduce computation time.").format(round(total_time,1)))
```
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/submission_viewer.py]: Actually once this message is displayed the statistics are computed and displayed too...so it is quite useless.

File: inginious/frontend/courses.py (hunk 1 above)
Comment (BIG-COW) [reply inginious/frontend/courses.py]: Yes it should, the translations are performed later in the HTML (submission_viewer.html, line 93).

**Comment by BIG-COW** [comment]
Change request is done.

**Pull request merged**